client/lib/feed-stream-store/paged-stream.js

```diff
@@ -248,11 +244,23 @@ assign( PagedStream.prototype, {
 			} );
 			this.postKeys = this.postKeys.concat( postKeys );
 			this.page++;
-			this.emit( 'change' );
+			this.emitChange();
 		}
 	}
-} );
 
-Emitter( PagedStream.prototype );
+	dismissPostsFromSite( blogId ) {
+		const removed = remove( this.postKeys, { blogId } );
+		if ( removed.length === 0 ) {
+			return; // nothing changed
+		}
+		// reshuffle what we have
+		this.postKeys = shuffle( this.postKeys );
```

Comment:
@blowery, when a user dismisses a rec, do we want to shuffle all the recs?  that seems counterintuitive to me!  I would expect:
1. user clicks X on left rec
2. rec gets removed & is replaced by a random rec thats left (`_.sample()`)

that way the order of recs earlier in the page is preserved, as well as the rec adjacent to the dismissed one